tiles/src/main/java/com/protomaps/basemap/layers/Roads.java

```diff
@@ -31,29 +49,91 @@ public void processFeature(SourceFeature sourceFeature, FeatureCollector feature
         .setAttrWithMinzoom("bridge", sourceFeature.getString("bridge"), 12)
         .setAttrWithMinzoom("tunnel", sourceFeature.getString("tunnel"), 12)
         .setAttrWithMinzoom("layer", sourceFeature.getString("layer"), 12)
-        .setAttrWithMinzoom("oneway", sourceFeature.getString("oneway"), 14)
-        .setAttr("ref", sourceFeature.getString("ref"));
+        .setAttrWithMinzoom("oneway", sourceFeature.getString("oneway"), 14);
 
       if (highway.equals("motorway") || highway.equals("motorway_link")) {
         feat.setAttr("pmap:kind", "highway").setZoomRange(6, 15);
-        OsmNames.setOsmNames(feat, sourceFeature, 10);
+
+        if (highway.equals("motorway")) {
+          feat.setAttrWithMinzoom("ref", shield_text, 7)
+            .setAttrWithMinzoom("ref_length", shield_text_length, 7)
```

Comment:
@bdon up to you what you want to call the exported `shield_text_length`. In Tilezen it's `shield_text_length` but other schemas call it `ref_length`.

Reply:
+1 to matching tilezen